test/metrics/diffusion/test_losses.py

```diff
@@ -0,0 +1,265 @@
+    loss_func = VPLoss()
+    assert loss_func.beta_d == 19.9
+    assert loss_func.beta_min == 0.1
+    assert loss_func.epsilon_t == 1e-5
+
+    loss_func = VPLoss(beta_d=10.0, beta_min=0.5, epsilon_t=1e-4)
+    assert loss_func.beta_d == 10.0
+    assert loss_func.beta_min == 0.5
+    assert loss_func.epsilon_t == 1e-4
+
+
+def test_sigma_method():
+    loss_func = VPLoss()
+
+    # Scalar input
+    sigma_val = loss_func.sigma(1.0)
+    assert isinstance(sigma_val, torch.Tensor)
+    assert sigma_val.item() > 0
+
+    # Tensor input
+    t = torch.tensor([1.0, 2.0])
+    sigma_vals = loss_func.sigma(t)
+    assert sigma_vals.shape == t.shape
+
+
+def fake_net(y, sigma, labels, augment_labels=None):
+    return torch.tensor([1.0])
+
+
+def test_call_method_vp():
+    loss_func = VPLoss()
+
+    images = torch.tensor([[[[1.0]]]])
+    labels = None
+
+    # Without augmentation
+    loss_value = loss_func(fake_net, images, labels)
+    assert isinstance(loss_value, torch.Tensor)
+
+    # With augmentation
+    def mock_augment_pipe(imgs):
+        return imgs, None
+
+    loss_value_with_augmentation = loss_func(
+        fake_net, images, labels, mock_augment_pipe
+    )
+    assert isinstance(loss_value_with_augmentation, torch.Tensor)
+
+
+# VELoss tests
+
+
+def test_veloss_initialization():
+    loss_func = VELoss()
+    assert loss_func.sigma_min == 0.02
+    assert loss_func.sigma_max == 100.0
+
+    loss_func = VELoss(sigma_min=0.01, sigma_max=50.0)
+    assert loss_func.sigma_min == 0.01
+    assert loss_func.sigma_max == 50.0
+
+
+def test_call_method_ve():
+    loss_func = VELoss()
+
+    images = torch.tensor([[[[1.0]]]])
+    labels = None
+
+    # Without augmentation
+    loss_value = loss_func(fake_net, images, labels)
+    assert isinstance(loss_value, torch.Tensor)
+
+    # With augmentation
+    def mock_augment_pipe(imgs):
+        return imgs, None
+
+    loss_value_with_augmentation = loss_func(
+        fake_net, images, labels, mock_augment_pipe
+    )
+    assert isinstance(loss_value_with_augmentation, torch.Tensor)
+
+
+# EDMLoss tests
+
+
+def test_edmloss_initialization():
+    loss_func = EDMLoss()
+    assert loss_func.P_mean == -1.2
+    assert loss_func.P_std == 1.2
+    assert loss_func.sigma_data == 0.5
+
+    loss_func = EDMLoss(P_mean=-2.0, P_std=2.0, sigma_data=0.3)
+    assert loss_func.P_mean == -2.0
+    assert loss_func.P_std == 2.0
+    assert loss_func.sigma_data == 0.3
+
+
+def test_call_method_edm():
+    loss_func = EDMLoss()
+
+    img = torch.tensor([[[[1.0]]]])
+    labels = None
+
+    # Without augmentation
+    loss_value = loss_func(fake_net, img, labels)
+    assert isinstance(loss_value, torch.Tensor)
+
+    # With augmentation
+    def mock_augment_pipe(imgs):
+        return imgs, None
+
+    loss_value_with_augmentation = loss_func(fake_net, img, labels, mock_augment_pipe)
+    assert isinstance(loss_value_with_augmentation, torch.Tensor)
+
+
+# RegressionLoss tests
```

Comment:
Maybe delete this?

Reply:
Thanks Oliver for your comment. Those tests are work in progress and I should be able to complete and uncomment those in a subsequent PR.